### ISSUE TITLE
Parameter only used in Metadata flagged

### ISSUE THREAD

**Issue by wjvanroosmalen** [aws-cloudformation/cfn-python-lint#1233]
*cfn-lint version: (`cfn-lint --version`)*
cfn-lint 0.25.5

*Description of issue.*
incorrect error, mentioning a parameter that is not used while it is used

*Additional info
```
  SourceEventTagFilter:
    Type: String
.....
Metadata:
  AWS::CloudFormation::Interface:
    ParameterGroups:
    - Parameters:
      - SourceEventTagFilter
      Label:
        default: Event source filtering
```

It seems that it does not see the resource when used as a parameter

**Comment by kddejong** [comment]
@wjvanroosmalen in your example are you saying the Metadata section is using the Parameter?  You are correct as we aren't checking Metadata to see when a Parameter is used.  That being said is there a valid reason for having a Parameter and only mentioning it in the Metadata section?  It still seems like in this scenario we aren't using the parameter still.

**Comment by wjvanroosmalen** [comment]
It is A CF template that is a part of multiple CF deployments. I use the parameters as a setting for a change in a DynamoDB table made by a Lambda. The Lambda takes the metadata from the CF template. As a kind op parameter store solution but it gives more flexibility (especially cross-region)

Maybe its a niche situation and I understand why you would want to alarm users using more typical CF templates on this.

**Comment by kddejong** [comment]
Yea I'm not sure I want ot change how that rule works but I do think there are a few options here.
- We create a parameter for that rule so you can specify to ignore that rule.  This rule has come up a few times so I think there could be some value here for the work.  There are other dynamic template items that do similar things.
- There is a long standing request to allow a YAML developer to specify exceptions with comments.  We haven't gotten far with that request but this has an outstanding issue.

For now you could also disable this rule just for that template to clean it up.
